Assignment4.java

```diff
@@ -0,0 +1,117 @@
+public class Assignment4 {
+    /**
+     * Finds the number of islands in the map using recursion.
+     *
+     * @param rowNumber    number of rows
+     * @param columnNumber number of columns
+     * @param map          map of the area
+     * @return number of islands, 0 if map is null
+     */
+    public static int findNumberOfIslandsRecursive(int rowNumber, int columnNumber, boolean[][] map) {
+        if (map == null) {
+            return 0;
+        }
+
+        int numberOfIslands = 0;
+        boolean[][] visited = new boolean[rowNumber][columnNumber];
+        boolean isLand;
```

Comment:
Small comment: variables should be declared as close as possible to the place where they are needed. In this case it should go into the innermost for loop: `boolean isLand = map[i][j];`